test/detach_on_cancel_test.cpp

```diff
@@ -162,3 +163,12 @@ TEST_F(detach_on_cancel_test, cancellation_and_completion_race) {
 
   EXPECT_EQ(max_iterations, count.load());
 }
+
+TEST_F(detach_on_cancel_test, error_types_propagate) {
+  using namespace unifex;
+  using error_types =
+    sender_error_types_t<decltype(detach_on_cancel(just())), type_list>;
+  using v = typename error_types::template apply<std::variant>;
```

Comment:
In the `broken-stdlib` branch, we're generally using `unifex:: variant` instead of `std::variant`; can't decide if it matters here, and I'm not sure whether we've got `unifex::variant_size` for the assertion, below.

```suggestion
  using v = typename error_types::template apply<unifex::variant>;
```